components/button/button-mixin.js

```diff
@@ -65,12 +65,36 @@ export const ButtonMixin = superclass => class extends FocusVisiblePolyfillMixin
 
 	constructor() {
 		super();
-		this.autofocus = false;
 		this.disabled = false;
-		this.primary = false;
+
+		/** @ignore */
+		this.autofocus = false;
+
+		/** @ignore */
 		this.type = 'button';
 	}
 
+	/**
```

Comment:
So this is the most annoying part. This is the only way I was able to get mixin attributes to "work" and show up in `custom-elements.json` with all the necessary pieces of information (e.g., default value, description, actual attribute name). `disabled` technically doesn't need this, it also works to just add the description above the `this.disabled = false` line in the constructor, but I thought I'd include both for consistency. If we did that though I believe it doesn't get included in the `attributes` array in `custom-elements.json`, which we aren't currently using (there's another array called members that is more detailed that I picked to use instead), but if we ever wanted to switch then we might want to not have to worry about some attributes not being present.

Reply:
That's... unfortunate. I wonder if we were using the TypeScript syntax for components if it'd pick those up better...

Reply:
Based on some examples I was looking at I think that it would.